include/swift/Basic/LangOptions.h

```diff
@@ -235,7 +236,11 @@ namespace swift {
 
     /// Diagnose uses of NSCoding with classes that have unstable mangled names.
     bool EnableNSKeyedArchiverDiagnostics = true;
-    
+
+    /// Regex for the passes that should report passed and missed optimizations.
+    std::shared_ptr<llvm::Regex> OptimizationRemarkPassedPattern;
```

Comment:
Why does this need to be a shared pointer. Can it be a unique pointer?

Reply:
On another note, how fast is llvm::Regex? IIRC @benlangmuir had to rip out the c++ regex since it was slow. I am assuming this is faster than that?

Reply:
I am not sure if it's faster or not but it only gets matched once per pass (not per remark!), so I don't think it should matter.

Reply:
Ah ok. But what about the unique_ptr?

Reply:
I am fixing that.  That's a left-over from clang...

Reply:
I am back to looking at this.  Changing to unique_ptr is not trivial as LangOpts is a member of CompilerInvocation which gets [copied into CompilerInstance](https://github.com/apple/swift/blob/62d7d79bb132d0b444f57eaacb4878bb322c4815/lib/Frontend/Frontend.cpp#L75).  Thus everything needs to be copy-assignable while unique_ptr is not.

Moving rather than copying CompilerInvocation is not an option because it's used [later](https://github.com/apple/swift/blob/62d7d79bb132d0b444f57eaacb4878bb322c4815/lib/FrontendTool/FrontendTool.cpp#L1419).

Using Regex directly does not work either as it's not copyable.

I presume clang used shared_ptr for similar reason.  I should obviously add a comment explaining this...